### ISSUE TITLE
.NET Core January 2018 Update

### ISSUE THREAD

**Issue by leecow** [dotnet/core#1199]
# .NET Core January 2018 Update

The .NET Core January 2018 Update is available. This release includes: 

* .NET Core 1.0.9, 1.1.6 and .NET Core SDK 1.1.7
* .NET Core 2.0.5 and .NET Core SDK 2.1.4

You can read about the updates in the [.NET Core release notes](https://github.com/dotnet/core/blob/master/release-notes/README.md).

Please report any issues you find with the January 2018 Update, either responding to this issue, creating a new issue or creating a new issue in one of the following repos:

* [dotnet/cli](https://github.com/dotnet/cli/issues) - for CLI tools and questions
* [dotnet/corefx](https://github.com/dotnet/corefx/issues) - for API issues and questions
* [dotnet/coreclr](https://github.com/dotnet/coreclr/issues) - for runtime issues
* [dotnet/core-setup](https://github.com/dotnet/core-setup/issues) - for installer issues
* [nuget/home](https://github.com/nuget/home/issues) - for NuGet questions and issues

**Comment by nulltoken** [comment]
⚠️ *This may not be the best place to report this. I'd be happy to open a new issue in another repository would anyone point me in the right direction.*

-------------

Is there any timeline to see this update deployed on **VSTS Hosted VS2017 agents**?

Updating the `global.json` file with the new version leads to such errors in the build logs.

```
2018-01-11T15:46:38.5191512Z Building...

2018-01-11T15:46:38.9108647Z ##[error]The specified SDK version [2.1.4] from global.json [D:\a\1\s\global.json] not found; install specified SDK version

Did you mean to run dotnet SDK commands? Please install dotnet SDK from:

  http://go.microsoft.com/fwlink/?LinkID=798306&clcid=0x409

 

2018-01-11T15:46:38.9133781Z

2018-01-11T15:46:40.8355697Z ##[error]An error occured...
```

Taking a look at the latest published content of the **[Hosted VS2017 agent software inventory](https://github.com/Microsoft/vsts-image-generation/blob/master/images/win/Vs2017/Vs2017-Server2016-Readme.md)**, it looks like the latest deployed version of .NET Core is `2.0.3`.

**Comment by Petermarcu** [comment]
@livarcocc may be able to get the right people to answer that. Are you using global.json because you want to force yourself to a specific SDK and not just use the latest that VSTS has?

**Comment by Chris197** [comment]
If I want to apply this new version to an existing installation using IIS on Windows Server, do I need to install the Windows Hosting Bundle or is that only required on first setup? Sorry if this has been asked before, I couldn't find the answer.

**Comment by livarcocc** [comment]
@joeloff to comment on the hosting bundle. But if you are trying to use the SDK on Windows Server, I would say it should be ok to just install the latest released SDK.

I will also try to find the right person regarding the version in VSTS. @leecow in case he can help here.

**Comment by joeloff** [comment]
You should consider updating the hosting bundle as the new one carries the package store for 2.0.5 as well as the 2.0.5 runtime. Additionally it has a newer version of ANCM for IIS as well. The hosting bundle though doesn't carry the SDK, so if you had a previous version of the SDK on the machine you'll probably want to update that separately.

Are you using the machine as a development environment? VS carries a copy of ANCM as well, but because the changes in ANCM for this release impacts hosting applications in IIS, VS was not updated with a new copy of ANCM.

**Comment by nulltoken** [comment]
> I will also try to find the right person regarding the version in VSTS. @leecow in case he can help here.

@Petermarcu @livarcocc Thanks!

**Comment by Chris197** [comment]
@joeloff , @livarcocc, Not using the machine for development, so I will install the hosting bundle to make sure all is up to date, thanks!

**Comment by TomCJones** [comment]
just tried to host a web app - it seems to always use port 5000 no matter what port I put in launchsettings.json. I tried to different solutions, one brand new.

**Comment by Petermarcu** [comment]
@Eilon @DamianEdwards

**Comment by Eilon** [comment]
@TomCJones - can you share more info about your setup? If you could upload a repro app to GitHub that would be ideal, but otherwise please share your launchSettings.json, appsettings.json, Program.cs, Startup.cs, and other relevant files. It could be that an environment variable or other config setting is overriding your desired port value.

**Comment by TomCJones** [comment]
sorry for the confusion, I didn't read the instructions carefully enough. They are kind of dense.

**Comment by bording** [comment]
I originally posted this in the SDK repo (https://github.com/dotnet/sdk/issues/1849), but it appears that there was a regression on some of the bundled templates in 2.14.

2.1.3
- microsoft.dotnet.common.itemtemplates.1.0.0-beta3-20171117-314.nupkg
- microsoft.dotnet.test.projecttemplates.2.0.1.0.0-beta3-20171117-314.nupkg

2.1.4
- microsoft.dotnet.common.itemtemplates.1.0.0-beta3-20171110-312.nupkg
- microsoft.dotnet.test.projecttemplates.2.0.1.0.0-beta3-20171110-312.nupkg

Was this intentional?

**Comment by livarcocc** [comment]
@mlorbetske to confirm the version that should be used!

**Comment by mlorbetske** [comment]
While this wasn't an intentional change (the move back to 312), the only impact here should be that the root namespace wouldn't be included in vbproj files for test projects (as it would have been introduced in build 313 which failed due to some CI issues which were corrected with 314) - the item templates should contain no differences.

**Comment by christiansk** [comment]
I installed the updates and can see version 2.0.5 in the command prompt using "dotnet --version", but VS2017 still shows the previous SDK version in my projects. I can see package Microsoft.NETCore.App 2.0.5 in NuGet but it won't let me update to saying the version is "blocked by project". How can I upgrade my project in VS to develop against the latest version?

**Comment by Petermarcu** [comment]
@leecow are the runtime and SDK versions mismatched in the original post? 

```
.NET Core 1.0.9, 1.1.6 and .NET Core SDK 2.1.4
.NET Core 2.0.5 and .NET Core SDK 1.1.7
```

I think it should be 2.X with 2.X and 1.X with 1.X.

**Comment by Petermarcu** [comment]
@christiansk , I believe you need the SDK version 2.1.4 in order to have VS support targeting 2.0.5. `dotnet --version` should say 2.1.4 if you have the right SDK installed.

**Comment by christiansk** [comment]
@Petermarcu You are correct, it says 2.1.4 - I confused the SDK/runtime versions when I wrote my post. Sorry about that. Still, I have it installed and can't get VS to allow me to update the SDK dependency. Any clue as to what else could cause the Microsoft.NETCore.App package to be "locked" to 2.0.0?

**Comment by Petermarcu** [comment]
Here are a few things to note.

1) By just installing 2.0.5, all your tools and applications that were running on shared 2.0.0 runtime will now run on 2.0.5.
2) 2.0.5 is not "targetable" per se. It doesn't have any new API's, just bug fixes. 

So just by installing the 2.1.4 SDK, when you do things like `dotnet run` a 2.0.0 application, it will automatically run on 2.0.5.

If you are doing a self-contained application, you need to explicitly set the version of the runtime you want with your application. This can be done by adding `<RuntimeFrameworkVersion>2.0.5</RuntimeFrameworkVersion>` to your project in a PropertyGroup.

Hope this helps.

**Comment by christiansk** [comment]
Got it. Thanks!

**Comment by leecow** [comment]
@Petermarcu , @christiansk - sorry about the version scrambling. Fixed!

**Comment by cgountanis** [comment]
So 2.1.4 was removed and replaced with 2.1.101?

**Comment by livarcocc** [comment]
2.1.101 is now the latest release of the .NET Core SDK.

You can still obtain the 2.1.4 from the all downloads link here: https://www.microsoft.com/net/download/all.

**Comment by cgountanis** [comment]
It is not there, I don;t need it just wondering why the version went down on the SDK and up on the runtime?

![image](https://user-images.githubusercontent.com/14899817/37416026-ce818544-277a-11e8-80dd-5031a9109ce4.png)

**Comment by Petermarcu** [comment]
Because we are realigning major and minor for the runtime and sdk.

**Comment by DamianEdwards** [comment]
2.1.101 is a higher version than 2.1.4
The numbers in each part are whole numbers, not fractions, so the 101 is > 4

**Comment by cgountanis** [comment]
Dah, thank you. 🤯

**Comment by Petermarcu** [comment]
Closing this one in favor of the latest announcement: https://github.com/dotnet/core/issues/1341
